backend/__tests__/cronJobs.test.ts

```diff
@@ -0,0 +1,122 @@
+import { expect } from 'chai';
+import mongoose from 'mongoose';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import ComplianceReport from '../src/db/schemas/compliance';
+import { startCronJobs } from '../src/cronJobs';
+import * as removeExpiredDrafts from '../src/cronJobs/removeExpiredDrafts';
+import { appConfig } from '../src/config';
+import sinon from 'sinon';
+
+describe('Cron Job', () => {
+    let mongod;
+    let clock;
+    let originalCronSchedule = appConfig.cron.removeExpiredDraftsSchedule;
+
+    before(async () => {
+        mongod = await MongoMemoryServer.create();
+        const uri = mongod.getUri();
+
+        if (mongoose.connection.readyState === 1) {
+            await mongoose.disconnect();
+        }
+        await mongoose.connect(uri);
+
+        const currentDate = new Date();
+        const expiredDate = new Date(currentDate);
+        expiredDate.setDate(expiredDate.getDate() - 1);
+        const futureDate = new Date(currentDate);
+        futureDate.setDate(futureDate.getDate() + 1);
+
+        await ComplianceReport.create({
+            softwareName: "openIMIS",
+            logo: "https://openIMIS.com/logo.png",
+            website: "https://openIMIS.com",
+            documentation: ["https://openIMIS.com/doc1", "https://openIMIS.com/doc2"],
+            pointOfContact: "contact@openIMIS.com",
+            status: 0,
+            expirationDate: new Date("2024-11-24T00:00:00.000Z"),
+            compliance: [],
+            link: "12345"
+        });
+        
+        await ComplianceReport.create({
+            softwareName: "openIMIS",
+            logo: "https://openIMIS.com/logo.png",
+            website: "https://openIMIS.com",
+            documentation: ["https://openIMIS.com/doc1", "https://openIMIS.com/doc2"],
+            pointOfContact: "contact@openIMIS.com",
+            status: 0,
+            expirationDate: new Date("2022-11-24T00:00:00.000Z"),
+            compliance: [],
+            link: "123456"
+        });
+
+        clock = sinon.useFakeTimers(currentDate.getTime());
+        appConfig.cron.removeExpiredDraftsSchedule = '* * * * * *';
+        startCronJobs();
+    });
+
+    after(async () => {
+        if (clock) {
+            clock.restore();
+        }
+        appConfig.cron.removeExpiredDraftsSchedule = originalCronSchedule;
+        await mongoose.connection.dropDatabase();
+        await mongoose.disconnect();
+        await mongod.stop();
+    });
+
+    it('should execute the cron job and call removeSensitiveDataFromExpiredDrafts multiple times', async () => {
+        const stub = sinon.stub(removeExpiredDrafts, 'removeSensitiveDataFromExpiredDrafts');
+        
+        for (let i = 0; i < 5; i++) {
+            await clock.tickAsync(1000);
+        }
+
+        expect(stub.callCount).to.be.greaterThan(1);
+        stub.restore();
+    });
```

Comment:
can you provide more details how this part works?

Reply:
It is designed to check whether a specific function is being called multiple times as a result of a cron job execution.
A Sinon stub is created for the `removeSensitiveDataFromExpiredDrafts` function, allowing the test to track how many times this function is called.  The fake clock is advanced by 5 seconds. During the time advancement, any cron jobs configured to run within this time frame are triggered, including potentially the one calling removeSensitiveDataFromExpiredDrafts.

 I will change it to check if the value is equal or greater than "5".